src/internet_identity/src/storage.rs

```diff
@@ -256,20 +271,17 @@ impl<T: candid::CandidType + serde::de::DeserializeOwned, M: Memory> Storage<T,
     }
 
     /// Make sure all the required metadata is recorded to stable memory.
-    pub fn flush(&self) {
-        if self.memory.size() < 1 {
```

Comment:
I'm not very familiar with this code; why don't we gros the memory anymore?

EDIT: I see, we're now using a writer. Why does the archive require that?